lib/filter/ignore.js

```diff
@@ -7,7 +7,12 @@ const matchToRule = require('../match').matchToRule;
 // given an ignore ruleset (parsed from the .snyk yaml file) and a array of
 // vulnerabilities, return the vulnerabilities that *are not* ignored
 // see http://git.io/vCHmV for example of what ignore structure looks like
-function filterIgnored(ignore, vuln, filtered) {
+function filterIgnored(
+  ignore,
+  vuln,
+  filtered,
+  matchStrategy = 'packageManager'
```

Comment:
"packageManager" as a `matchStrategy` sounds weird no?

Reply:
@darscan I can see that, yeah. But I'm not sure how else to name the set of characteristics that I've enumerated in https://github.com/snyk/policy/pull/58#discussion_r682458270 and the PR description. The new IaC code path could be paraphrased "match this like a file-qualified jsonpath", and the existing way could be paraphrased like "match this like a node in a package manager's graph".